cartographer/transform/transform.h

```diff
@@ -51,31 +51,31 @@ T GetYaw(const Rigid3<T>& transform) {
 // rotation as the given 'quaternion'.
 template <typename T>
 Eigen::Matrix<T, 3, 1> RotationQuaternionToAngleAxisVector(
-    Eigen::Quaternion<T> quaternion) {
-  quaternion.normalize();
+    const Eigen::Quaternion<T>& quaternion) {
+  auto normalized_quat = quaternion.normalized();
```

Comment:
Consistency: Using `auto` with Eigen is problematic ([expression templates](https://en.wikipedia.org/wiki/Expression_templates), see https://eigen.tuxfamily.org/dox/TopicPitfalls.html) so it is best avoided. Prefer `Eigen::Quaternion<T> normalized_quaternion = ...`

Style: Avoid abbreviatons. `normalized_quaternion`?